qadence/backends/parameter_shift_rules.py

```diff
@@ -93,30 +109,39 @@ def multi_gap_psr(
     param_dict: dict[str, Tensor],
     param_name: str,
     spectral_gaps: Tensor,
-    shift_prefac: float = 0.5,
+    shift_prefac: float | None = 0.5,
+    lb: float | None = None,
+    ub: float | None = None,
 ) -> Tensor:
     """Implements multi-gap multi-qubit GPSR rule.
 
     Args:
         expectation_fn (Callable[[dict[str, Tensor]], Tensor]): backend-dependent function
         to calculate expectation value
-
         param_dict (dict[str, Tensor]): dict storing parameters values of parameterized blocks
         param_name (str): name of parameter with respect to that differentiation is performed
         spectral_gaps (Tensor): tensor containing spectral gap values
         shift_prefac (float): prefactor governing the magnitude of parameter shift values -
         select smaller value if spectral gaps are large
+        lb (float): lower bound of optimal shift value search interval
+        ub (float): upper bound of optimal shift value search interval
 
     Returns:
         Tensor: tensor containing derivative values
     """
     n_eqs = len(spectral_gaps)
     batch_size = max(t.size(0) for t in param_dict.values())
 
-    # get shift values
-    shifts = shift_prefac * torch.linspace(
-        PI / 2 - PI / 4, PI / 2 + PI / 5, n_eqs
-    )  # breaking the symmetry of sampling range around PI/2
+    # get shift values - values minimize the variance of expectation
+    if shift_prefac is not None:
+        # set shift values manually
+        shifts = shift_prefac * torch.linspace(
+            PI / 2 - PI / 4, PI / 2 + PI / 5, n_eqs
```

Comment:
```suggestion
            PI / 4, 7 * PI / 10, n_eqs
```
It would be good to elaborate a little more on where these numbers come from.